Error previewing update in azure-ts-aks-helm

_(Newbie disclaimer)_

I'm trying this example: https://github.com/pulumi/examples/tree/master/azure-ts-aks-helm

I click on the 'Deploy with Pulumi' button and the stack gets created ok in my project.

When I attempt the `pulumi new` command 

```
pulumi new https://github.com/pulumi/examples/tree/master/azure-ts-aks-helm -s <MyStack>
```
I get this error:

> Running program 'C:\src\pulumi\azure-ts-aks-helm-dev' failed with an unhandled exception:
> TSError: ⨯ Unable to compile TypeScript:
> index.ts(21,6): error TS2339: Property 'status' does not exist on type 'Output<Service>'.
> index.ts(21,19): error TS7006: Parameter 'status' implicitly has an 'any' type.
> 
>     at createTSError (C:\src\pulumi\azure-ts-aks-helm-dev\node_modules\ts-node\src\index.ts:261:12)
>     at getOutput (C:\src\pulumi\azure-ts-aks-helm-dev\node_modules\ts-node\src\index.ts:367:40)
>     at Object.compile (C:\src\pulumi\azure-ts-aks-helm-dev\node_modules\ts-node\src\index.ts:558:11)
>     at Module.m._compile (C:\src\pulumi\azure-ts-aks-helm-dev\node_modules\ts-node\src\index.ts:439:43)
>     at Module._extensions..js (internal/modules/cjs/loader.js:733:10)
>     at Object.require.extensions.(anonymous function) [as .ts] (C:\src\pulumi\azure-ts-aks-helm-dev\node_modules\ts-node\src\index.ts:442:12)
>     at Module.load (internal/modules/cjs/loader.js:620:32)
>     at tryModuleLoad (internal/modules/cjs/loader.js:560:12)
>     at Function.Module._load (internal/modules/cjs/loader.js:552:3)
>     at Module.require (internal/modules/cjs/loader.js:658:17)
> an unhandled error occurred: Program exited with non-zero exit code: 1

I'm on Windows 10 and my $PSVersionTable is:

> PSVersion                      5.1.17134.407
> PSEdition                      Desktop
> PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
> BuildVersion                   10.0.17134.407
> CLRVersion                     4.0.30319.42000
> WSManStackVersion              3.0
> PSRemotingProtocolVersion      2.3
> SerializationVersion           1.1.0.1

Any suggestions?

Many Thanks

--- a/azure-ts-aks-helm/index.ts
+++ b/azure-ts-aks-helm/index.ts
@@ -17,6 +17,6 @@ const apache = new helm.v2.Chart(
 export let cluster = k8sCluster.name;
 export let kubeConfig = k8sCluster.kubeConfigRaw;
 export let serviceIP = apache
-    .getResource("v1/Service", "apache-apache")
-    .status.apply(status => status.loadBalancer.ingress[0].ip);
+    .getResourceProperty("v1/Service", "apache-apache", "status")
+    .apply(status => status.loadBalancer.ingress[0].ip);
 

--- a/azure-ts-aks-mean/index.ts
+++ b/azure-ts-aks-mean/index.ts
@@ -55,5 +55,5 @@ const node = new k8s.helm.v2.Chart(
 export const kubeconfig = k8sCluster.kubeConfig;
 export const cluster = k8sCluster.name;
 export const frontendAddress = node
-    .getResource("v1/Service", "node-node")
-    .status.apply(status => status.loadBalancer.ingress[0].ip);
+    .getResourceProperty("v1/Service", "node-node", "status")
+    .apply(status => status.loadBalancer.ingress[0].ip);

--- a/kubernetes-ts-helm-wordpress/index.ts
+++ b/kubernetes-ts-helm-wordpress/index.ts
@@ -16,5 +16,5 @@ const wordpress = new k8s.helm.v2.Chart("wpdev", {
 });
 
 // Export the public IP for Wordpress.
-const frontend = wordpress.getResource("v1/Service", "wpdev-wordpress");
-export const frontendIp = frontend.status.apply(status => status.loadBalancer.ingress[0].ip);
+const frontend = wordpress.getResourceProperty("v1/Service", "wpdev-wordpress", "status");
+export const frontendIp = frontend.apply(status => status.loadBalancer.ingress[0].ip);

--- a/kubernetes-ts-helm-wordpress/package.json
+++ b/kubernetes-ts-helm-wordpress/package.json
@@ -4,7 +4,7 @@
         "@types/node": "latest"
     },
     "dependencies": {
-        "@pulumi/pulumi": "^0.15.0",
-        "@pulumi/kubernetes": "^0.16.0"
+        "@pulumi/pulumi": "^0.16.7",
+        "@pulumi/kubernetes": "^0.18.0"
     }
 }

--- a/kubernetes-ts-staged-rollout-with-prometheus/util.ts
+++ b/kubernetes-ts-staged-rollout-with-prometheus/util.ts
@@ -4,6 +4,7 @@ import * as pulumi from "@pulumi/pulumi";
 import * as k8s from "@pulumi/kubernetes";
 import * as http from "http";
 import { spawn } from "child_process";
+import { meta } from "@pulumi/kubernetes/types/input";
 
 export interface PromPortForwardOpts {
     localPort: number;
@@ -19,15 +20,15 @@ export interface PromPortForwardOpts {
 // This is useful for when we can't run Pulumi in-cluster, in which case simply calling to the
 // appropriate KubeDNS URL is not sufficient.
 export function forwardPrometheusService(
-    service: k8s.core.v1.Service,
-    deployment: k8s.extensions.v1beta1.Deployment,
+    service: pulumi.Input<k8s.core.v1.Service>,
+    deployment: pulumi.Input<k8s.extensions.v1beta1.Deployment>,
     opts: PromPortForwardOpts
 ): pulumi.Output<() => void> {
     if (pulumi.runtime.isDryRun()) {
         return pulumi.output(() => {});
     }
 
-    return pulumi.all([service.metadata, deployment.urn]).apply(([meta]) => {
+    return pulumi.all([service, deployment]).apply(([s, d]) => pulumi.all([s.metadata, d.urn])).apply(([meta]) => {
         return new Promise<() => void>((resolve, reject) => {
             const forwarderHandle = spawn("kubectl", [
                 "port-forward",
@@ -182,7 +183,7 @@ function getHttpLatency(url: string): Promise<string> {
 
             // Append to the body until the end.
             const body: string[] = [];
-            response.on("data", chunk => body.push(chunk));
+            response.on("data", chunk => body.push(chunk.toString()));
             response.on("end", () => resolve(body.join("")));
         });
 

